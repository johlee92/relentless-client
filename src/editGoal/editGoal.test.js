import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EditGoal from './editGoal';
import { BrowserRouter } from 'react-router-dom';

describe(`EditGoal component`, () => {
    const props = {
        className: 'test-class-name',
        viewGoals: 'test-view-goals'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <EditGoal {...props}/>
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component from props', () => {
        const wrapper = shallow(<EditGoal {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
