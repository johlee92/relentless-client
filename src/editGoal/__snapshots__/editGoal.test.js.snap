// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGoal component renders the component from props 1`] = `
<div
  className="editGoal"
/>
`;
